### PR TITLE
Model by reference for Multiple Models

### PR DESCRIPTION
**Pull request by VipulMascarenhas**
### Description

This PR covers the following:
1. change `artifact` property to be set either as a string or a list using an iterable.
2. For Large Artifact Downloader with model by reference, set bucket_uri as None as artifact will be downloaded using model_file_description.
3. For model by reference, files are saved locally at `target_dir/bucket_name/file_name_with_prefix`.

### Tests

1. Creating model with multiple artifact location

```
artifact_path = "oci://service-bucket@namespace/model_path"
weights_path = "oci://user-bucket@namespace/weights_path"

model = (DataScienceModel()
        .with_compartment_id(compartment_id)
        .with_project_id(project_id)
        .with_display_name(MODEL_NAME)
        .with_artifact(artifact_path, weights_path)
        )

model.create(model_by_reference=True,)

model.download_artifact(target_dir="artifact_dir")
```

2. Creating model using yaml 

`test.yaml`
```
kind: datascienceModel
spec:
  compartmentId: ocid1.compartment.oc1..<OCID>
  projectId: ocid1.datascienceproject.oc1.iad.<OCID>
  artifact: [oci://service-bucket@namespace/model_path, oci://user-bucket@namespace/weights_path]
  displayName: Model Name
...
...
```

```
model = DataScienceModel.from_yaml(uri="test.yaml")
model.create(model_by_reference=True)
```

3. Loading model by id already created by reference with multiple models/artifact location

```
model = DataScienceModel.from_id("ocid1.datasciencemodel.oc1.iad.<OCID>")
model.download_artifact(target_dir="artifact_dir")
```

### CODE REVIEW

File: ads/model/datascience_model.py [hunk 1]
```diff
@@ -1245,14 +1267,23 @@ def __getattr__(self, item):
...
-            logger.error(message)
-            raise BucketNotVersionedError(message)
+    def _validate_prepare_file_description_artifact(self):
+        """This helper method validates the path to check if the buckets are versioned and if the OSS location and
+        the files exist. Next, it creates a json dict with the path information and sets it as the artifact to be
+        uploaded."""
+
+        bucket_uri = self.artifact
+        if isinstance(bucket_uri, str):
+            bucket_uri = list(bucket_uri)
```
Comment (qiuosier) [inline ads/model/datascience_model.py:1277]: Do you intend to do `bucket_uri = [bucket_uri]`? `list` will make a single string as a list of characters.

File: ads/model/datascience_model.py [hunk 2]
```diff
@@ -539,13 +544,17 @@ def with_artifact(self, uri: str):
             Path to artifact directory or to the ZIP archive.
             It could contain a serialized model(required) as well as any files needed for deployment.
             The content of the source folder will be zipped and uploaded to the model catalog.
-
+            For models created by reference, uri can take in single arg or multiple args in case of a fine-tuned or
+            multimodel setting.
         Examples
         --------
         >>> .with_artifact(uri="./model1/")
         >>> .with_artifact(uri="./model1.zip")
```
Comment (qiuosier) [inline ads/model/datascience_model.py:552]: `uri` can no longer be used as keyword argument once it is written as `*uri`. I think we can do `with_artifact(self, uri, *args)` to keep backward compatibility and enforce the requirement of at least one argument.

File: ads/model/datascience_model.py (hunk 1 above)
Comment (VipulMascarenhas) [reply ads/model/datascience_model.py:1277]: you're right - I had fixed that locally while testing, hadn't pushed it. Will make this change.

File: ads/model/datascience_model.py (hunk 2 above)
Comment (VipulMascarenhas) [reply ads/model/datascience_model.py:552]: yes, makes sense. Will update this.

**Review by qiuosier — APPROVED** [review]

**Pull request merged**